django/contrib/gis/db/models/lookups.py

```diff
@@ -79,14 +79,16 @@ def as_sql(self, compiler, connection):
         lhs_sql, lhs_params = self.process_lhs(compiler, connection)
         rhs_sql, rhs_params = self.process_rhs(compiler, connection)
         sql_params = (*lhs_params, *rhs_params)
+        rhs_op = self.get_rhs_op(connection, rhs_sql)
+        if self.lhs.output_field.geography and not rhs_op.geography:
+            lhs_sql += "::geometry"
```

Comment:
This will break for 3rd-party database backends that support `geography`. We could move it to the `DatabaseOperations.field_cast_sql()` like in the `BuiltinLookup.process_lhs()` :thinking: :

https://github.com/django/django/blob/5e9aded33f46dd78c5b6050186ba54ff9b02e986/django/db/models/lookups.py#L218-L220

Reply:
Ah, right, I need to stay in the PostGIS space. I ran across this comment, which pointed me toward `PostGISOperator.as_sql`, since I do need the "lhs part mixed in", and it was finding the right place that has knowledge of both lhs and rhs that was tricky.

https://github.com/django/django/blob/bd366ca2aeffa869b7dbc0b0aa01caea75e6dc31/django/contrib/gis/db/models/lookups.py#L73-L75

So I added `check_geometry` in XXXXX, and this formed a nice parallel with `check_raster`. What do you think?